diff --git a/gaia/build/settings.js b/gaia/build/settings.js
index 50ac96b..b0fbc98 100644
--- a/gaia/build/settings.js
+++ b/gaia/build/settings.js
@@ -291,20 +291,22 @@ function execute(config) {
   if (config.NO_LOCK_SCREEN === '1') {
     settings['lockscreen.enabled'] = false;
     settings['lockscreen.locked'] = false;
   }
 
   var screenTimeout = parseInt(config.SCREEN_TIMEOUT, 10);
   if (screenTimeout >= 0) {
     settings['screen.timeout'] = screenTimeout;
   }
 
+  settings['screen.timeout'] = 0;
+
   setDefaultKeyboardLayouts(config.GAIA_DEFAULT_LOCALE, settings, config);
 
   var queue = utils.Q.defer();
   queue.resolve();
 
   var result = queue.promise.then(function() {
     setWallpaper(settings, config);
   }).then(function() {
     setMediatone(settings, config);
   }).then(function() {
