diff --git a/gaia/apps/system/js/screen_manager.js b/gaia/apps/system/js/screen_manager.js
index 372ce1e..de7c7ae 100644
--- a/gaia/apps/system/js/screen_manager.js
+++ b/gaia/apps/system/js/screen_manager.js
@@ -547,7 +547,7 @@ var ScreenManager = {
 
     var self = this;
     var idleCallback = function idle_proxy() {
-      self.turnScreenOff(instant, 'idle_timeout');
+     // self.turnScreenOff(instant, 'idle_timeout');
     };
     var activeCallback = function active_proxy() {
       self.turnScreenOn(true);
